modules/imgcodecs/src/grfmt_sunras.cpp

```diff
@@ -60,6 +22,7 @@ SunRasterDecoder::SunRasterDecoder()
     m_encoding = RAS_STANDARD;
     m_maptype = RMT_NONE;
     m_maplength = 0;
+    m_buf_supported = true;
```

Comment:
Please extend existing test to cover the new feature.

Reply:
when `m_buf_supported = false;` imencode function writes the data into a temporary file. so tests already cover the functionality of `m_buf_supported = true;`